### PR TITLE
feat: add experiment summarizer for standard and nocturnal forecasting

### PR DESCRIPTION
**Pull request by RobotPsychologist**
Adds ExperimentSummarizer base class and two concrete implementations (StandardForecastingSummarizer, NocturnalSummarizer) that walk experiment run directories, build a tidy summary.csv, and surface the best run per model and per (model × dataset) pair.

- src/experiments/base/experiment.py: abstract base with summarize() and best_runs(), user-specifiable ranking metric (rmse/mae/mape/mse)
- src/experiments/standard_forecasting/summarize.py: parses results.json
- src/experiments/nocturnal/summarize.py: parses nocturnal_results.json
- scripts/analysis/summarize_experiments.py: CLI entry point
- tests/experiments/: 39 tests covering parsing, edge cases, and ranking

Writes summary.csv, best_by_model_dataset.csv, and best_by_model.csv into each experiment directory.

### CODE REVIEW

File: tests/experiments/test_nocturnal_summarizer.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        assert len(by_md) == 2
+        brown = by_md[by_md["dataset"] == "brown_2019"].iloc[0]
+        assert brown["rmse"] == pytest.approx(3.20)
+
+    def test_best_by_model(self, tmp_path):
+        self._populate(tmp_path)
+        result = NocturnalSummarizer(tmp_path).best_runs()
+        by_m = result["by_model"]
+        assert len(by_m) == 1
+        assert by_m.iloc[0]["rmse"] == pytest.approx(2.85)
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply tests/experiments/test_nocturnal_summarizer.py]: Added in next push.

File: src/experiments/nocturnal/summarize.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        overall_rmse = data.get("overall_rmse")
+        if overall_rmse is None:
+            log.debug("Missing 'overall_rmse' in %s — skipping", results_path)
+            return None
+
+        total_episodes = data.get("total_episodes")
+        if total_episodes is None:
+            # Fall back to summing per_patient episodes
+            per_patient = data.get("per_patient", [])
+            total_episodes = int(sum(p.get("episodes", 0) for p in per_patient))
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply src/experiments/nocturnal/summarize.py]: Added recommendation.

File: scripts/analysis/summarize_experiments.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    print(f"{'='*70}")
+
+    summarizer = cls(experiments_root=root)
+    results = summarizer.best_runs(metric=metric)
+
+    by_model_dataset = results["by_model_dataset"]
+    by_model = results["by_model"]
+
+    if by_model_dataset.empty:
+        print("  [No completed runs found]\n")
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply scripts/analysis/summarize_experiments.py]: Fixed in next run.

File: src/experiments/base/experiment.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        return df
+
+    def best_runs(
+        self,
+        metric: str = "rmse",
+    ) -> dict[str, pd.DataFrame]:
+        """Return (and save) the best run per model and per (model × dataset).
+
+        Calls :meth:`summarize` internally if not already run.
+
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply src/experiments/base/experiment.py]: updated.
